sdk/appconfiguration/app-configuration/test/internal/http.spec.ts

```diff
@@ -32,17 +33,36 @@ describe("http request related tests", () => {
     });
 
     it("useragentheadername", () => {
-      assert.ok(getUserAgentHeaderName({
+      let options = getGeneratedClientOptions("base-uri", new SyncTokens(), {
+        isNodeOverride: false
+      });
+
+      assert.equal(options.userAgentHeaderName, "x-ms-useragent", "Pretending we're running in a browser.");
+
+      options = getGeneratedClientOptions("base-uri", new SyncTokens(), {
         isNodeOverride: true
-      }) === undefined);
+      });
+
+      assert.equal(options.userAgentHeaderName, "User-Agent", "Pretending we're running in node.");
 
       // since we're only running these tests in node this will be the same as the 
       // case above (undefined, thus using the normal User-Agent header)
-      assert.ok(getUserAgentHeaderName({}) === undefined);
+      options = getGeneratedClientOptions("base-uri", new SyncTokens(), {});
 
-      assert.equal(getUserAgentHeaderName({
-        isNodeOverride: false
-      }), "x-ms-useragent");
+      assert.equal(options.userAgentHeaderName, "User-Agent", "We know that we're running node.");
+    });
+      
+    it("useragent", () => {
+      let options = getGeneratedClientOptions("base-uri", new SyncTokens(), {
+        userAgentOptions: {
+          userAgentPrefix: "MyCustomUserAgent"
+        }
+      });
+
+      chai.assert.match(options.userAgent as string, /^MyCustomUserAgent azsdk-js-app-configuration\/[^ ]+ core-http\/[^ ]+.+$/, `Using a custom user agent`);
```

Comment:
Thoughts on adding the package version into the regex so we can be sure it's the right version? You can do that if you use the `RegExp` class and can then pass in a string template so you don't have to update the version manually everytime we update the package.

Reply:
Done.